server/src/main/java/org/opensearch/action/get/TransportMultiGetAction.java

```diff
@@ -112,7 +112,7 @@ protected void doExecute(Task task, final MultiGetRequest request, final ActionL
 
             MultiGetShardRequest shardRequest = shardRequests.get(shardId);
             if (shardRequest == null) {
-                if (shouldForcePrimaryRouting(clusterState, request.realtime, request.preference, concreteSingleIndex)) {
+                if (shouldForcePrimaryRouting(clusterState.getMetadata(), request.realtime, request.preference, concreteSingleIndex)) {
```

Comment:
Lets pls remove the coupling of these two action classes.  This can implement the same check shouldForcePrimaryRouting